app/controllers/delivery_controller.rb

```diff
@@ -0,0 +1,48 @@
+ class DeliveryController < ApplicationController
+  def index
+    @matching_deliveries = Delivery.where({:user_id => current_user})
+    @pending = @matching_deliveries.where({:arrived => false })
+    @received = @matching_deliveries.where({:arrived => true})
+    render(:templates => 'delivery/index')
+  end
+
+  def show
+    delivery_id = params.fetch("id")
+    @deliveries = Delivery.where({:id => delivery_id})
+    render(:templates => 'delivery/show')
+  end
+
+  def create
+    delivery = Delivery.new
+    delivery.supposed_to_arrive_on = params.fetch("supposed_to_arrive_on")
+    delivery.description = params.fetch("description")
+    delivery.details = params.fetch("details")
+    delivery.created_at = Time.now
```

Comment:
These are automatically done when you create and update an object. No need to manually do it